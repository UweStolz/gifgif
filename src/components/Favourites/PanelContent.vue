<template>
  <v-expansion-panel-content id="pa-exp-con">
    <v-item-group>
      <v-row
        justify="start"
        align="start"
      >
        <Dialog
          :show-dialog.sync="dialog"
          :full-image-data.sync="fullImage"
          :image-id.sync="imageId"
          @delete="deleteGif"
        />
        <v-item
          v-for="(z, index) in previewImages.length"
          :key="z"
          class="mx-auto"
        >
          <v-lazy min-height="50">
            <v-img
              id="imageHover"
              :src="previewImages[index]"
              :lazy-src="previewImages[index]"

              max-width="230"
              max-height="200"
              @click="openDialog(index)"
            />
          </v-lazy>
        </v-item>
      </v-row>
    </v-item-group>
  </v-expansion-panel-content>
</template>

<script lang="ts">
import { Vue, Component, Prop } from 'vue-property-decorator';
import Dialog from './Dialog.vue';

@Component({
  components: {
    Dialog,
  },
})
export default class PanelContent extends Vue {
  @Prop({ required: true }) previewImages!: string[];

  @Prop({ required: true }) fullImages!: (Blob | string)[];

  @Prop({ required: true }) imageId!: string[];

  fullImage: null | Blob | string = null;

  dialog: boolean = false;

  selectedIndex: number = -1;

  openDialog(index: number) {
    this.dialog = true;
    this.fullImage = this.fullImages[index];
    this.selectedIndex = index;
  }

  async deleteGif(payload: string): Promise<void> {
    await this.$store.dispatch('removeGifData', payload[this.selectedIndex]);
    this.$store.commit('setGifCount', this.$store.state.gifCount - 1);
    this.previewImages.splice(this.selectedIndex, 1);
    this.fullImages.splice(this.selectedIndex, 1);
    this.imageId.splice(this.selectedIndex, 1);
    this.selectedIndex = -1;
  }
}
</script>

<style scoped>
#pa-exp-con {
  overflow: hidden;
}
#imageHover:hover {
  transform: scale(1.3);
  overflow-x: hidden;
  z-index: 999;
}
</style>
