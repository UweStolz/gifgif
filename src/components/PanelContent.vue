<template>
  <v-expansion-panel-content>
    <v-item-group>
      <v-container>
        <v-row
          justify="start"
          align="start"
        >
          <v-item
            v-for="(z, index) in contentData.length"
            :key="z"
          >
            <v-img
              :src="contentData[index]"
              contain
              max-width="200"
              max-height="200"
            />
          </v-item>
        </v-row>
      </v-container>
    </v-item-group>
  </v-expansion-panel-content>
</template>

<script lang="ts">
import { Vue, Component, Prop } from 'vue-property-decorator';

@Component
export default class PanelContent extends Vue {
@Prop({ required: true }) contentData: any

indexValue: number = 0;
}
</script>
