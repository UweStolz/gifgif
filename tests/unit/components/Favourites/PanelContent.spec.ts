import PanelContent from '@/components/Favourites/PanelContent.vue';
import shallow from '../../../helper';

describe('PanelContent.vue', () => {
  it('renders properly', () => {
    const wrapper = shallow(PanelContent, {
      propsData: {
        imageIds: ['id-1', 'id-2'],
        previewImages: ['previewImage1', 'previewImage2'],
        fullImages: ['fullImage1', 'fullImage2'],
      },
    });
    expect(wrapper).toMatchSnapshot();
  });
});
