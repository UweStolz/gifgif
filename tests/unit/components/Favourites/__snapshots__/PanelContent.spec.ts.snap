// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PanelContent.vue renders properly 1`] = `
<v-expansion-panel-content-stub id="pa-exp-con">
  <v-item-group-stub activeclass="v-item--active">
    <v-row-stub tag="div" align="start" justify="start">
      <dialog-stub></dialog-stub>
      <v-item-stub class="mx-auto">
        <v-lazy-stub minheight="50" options="[object Object]" tag="div" transition="fade-transition">
          <v-img-stub maxheight="200" maxwidth="230" lazysrc="previewImage1" options="[object Object]" position="center center" src="previewImage1" transition="fade-transition" id="imageHover"></v-img-stub>
        </v-lazy-stub>
      </v-item-stub>
      <v-item-stub class="mx-auto">
        <v-lazy-stub minheight="50" options="[object Object]" tag="div" transition="fade-transition">
          <v-img-stub maxheight="200" maxwidth="230" lazysrc="previewImage2" options="[object Object]" position="center center" src="previewImage2" transition="fade-transition" id="imageHover"></v-img-stub>
        </v-lazy-stub>
      </v-item-stub>
    </v-row-stub>
  </v-item-group-stub>
</v-expansion-panel-content-stub>
`;
